docs/data-sources/application.md

```diff
@@ -43,7 +44,8 @@ The following attributes are exported:
 * `api` - An `api` block as documented below.
 * `app_role_ids` - A mapping of app role values to app role IDs, intended to be useful when referencing app roles in other resources in your configuration.
 * `app_roles` - A collection of `app_role` blocks as documented below. For more information see [official documentation on Application Roles](https://docs.microsoft.com/en-us/azure/architecture/multitenant-identity/app-roles).
-* `application_id` - The Application ID (also called Client ID).
+* `application_id` - (Deprecated) The Client ID for the application.
```

Comment:
should we just remove this like we do in rm as its deprecated? / should be `deprecated in favour of X`